packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -201,6 +208,29 @@ function checkProofEntryConfig(
       entryConfig.equalsEntry
     );
   }
+
+  const hasBoundsCheck = entryConfig.inRange !== undefined;
+
+  if (hasBoundsCheck) {
+    const inRange = entryConfig.inRange as BoundsConfig;
```

Comment:
I think this could be `satisfies` rather than `as`.
As I understand it, the former will compile only if the input really does match the fields for right type, while the latter will "take your word for it" and not trigger a compile error if you did something wrong.

Reply:
Agreed, but weirdly enough the compiler still thinks it could be `undefined` so that `satisfies` does not work, which is why I introduced this variable to do the type coercion for the following lines. It is a different story with `if(hasBoundsCheck)` replaced with `if(entryConfig.inRange !== undefined)`.

Reply:
Oh, yeah I've seen this particular bit of dumb typechecking before.
I think if you saved the bounds in a variable and then checked that variable directly against undefined in your if statement the compiler would see that it's defined, but then you return statement would get more complicated.
I wish there were a more limited form for a compile-time assertion that something is defined, without fully overriding its type.